flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestUtils.java

```diff
@@ -77,10 +77,17 @@ public static List<Row> collectAllRows(Table table) throws Exception {
         final TableResult result = table.execute();
 
         final List<Row> collectedRows = new ArrayList<>();
-        try (CloseableIterator<Row> iterator = result.collect()) {
+        CloseableIterator<Row> iterator = result.collect();
+        try {
             while (iterator.hasNext()) {
                 collectedRows.add(iterator.next());
             }
+        } finally {
+            try {
+                iterator.close();
+            } catch (Exception e) {
+                // This partially solves FLINK-19204 for Kafka Tests
+            }
```

Comment:
I know it's only test code, but this pattern always catches an eye since we also catch exceptions like `InterruptedException` and we reset the interrupted flag.

Is this necessary to fix on this PR? I would much rather invest in fixing the cause than fighting the symptoms.

Reply:
The root cause is that closing the iterator is not graceful. But that's an issue in Flink 1.20 that can't be fixed without a new Flink release.
I'll try to pick a more specific exception though to avoid catching IE.

Reply:
I now specifically pass a list of allowed exceptions.

```
    private void cleanupTopic(String topic) {
        ignoreExceptions(() -> deleteTestTopic(topic), UnknownTopicOrPartitionException.class);
    }

    private static void ignoreExceptions(
            RunnableWithException e, Class<? extends Exception>... exClasses) {
        try {
            e.run();
        } catch (Exception ex) {
            assertThatChainOfCauses(ex)
                    .anyMatch(
                            cause -> Arrays.stream(exClasses).anyMatch(cl -> cl.isInstance(cause)));
        }
    }

    private static void cancelJob(TableResult tableResult) {
        if (tableResult != null) {
            ignoreExceptions(
                    () -> tableResult.getJobClient().ifPresent(JobClient::cancel),
                    FlinkJobNotFoundException.class,
                    FlinkJobTerminatedWithoutCancellationException.class);
        }
    }
```